### ISSUE TITLE
can not use angular 12 with latest version.

### ISSUE THREAD

**Issue by klerick** [nestjs/ng-universal#624]
Updating ng-universal v5.0.0 returns a strange error after trying to open angular page.

ng-universal v5.0.0 depend on nestjs 8 -> rxjs 7
but angular 12 using rxjs 6 now. 
so we have a multi-level tree with cross dependencies

if using ng-universal v4 with angular 12 and @nguniversal/* 12, we have a multi-level tree with cross dependencies too.

as a result builder can not resolve dependency correctly.

Can we create an intermediate version for angular 12 with nestjs 7 and rxjs 6?

**Comment by klerick** [comment]
@kamilmysliwiec Do you have a solution for this situation?

**Comment by WeslleyDeSouza** [comment]
> @kamilmysliwiec Do you have a solution for this situation?
try this one:<br>
```
"dependencies": {
  "@angular/animations": "~12.1.4",
    "@angular/common": "~12.1.4",
    "@angular/compiler": "~12.1.4",
    "@angular/core": "~12.1.4",
    "@angular/forms": "~12.1.4",
    "@angular/platform-browser": "~12.1.4",
    "@angular/platform-browser-dynamic": "~12.1.4",
    "@angular/platform-server": "~12.1.4",
    "@angular/router": "~12.1.4",
    "@nestjs/common": "^8.0.5",
    "@nestjs/core": "^8.0.5",
    "@nestjs/ng-universal": "^5.0.0",
    "@nestjs/passport": "^8.0.0",
    "@nestjs/platform-express": "^8.0.5",
    "@nguniversal/common": "12.1.0",
    "@nguniversal/express-engine": "12.1.0",
    "class-transformer": "^0.2.3",
    "class-validator": "^0.9.1",
    "classlist.js": "1.1.20150312",
    "core-js": "^2.6.5",
    "passport": "^0.4.1",
    "passport-local": "^1.0.0",
    "rxjs": "~6.6.7",
    "terser": "^4.1.2",
    "tsconfig-paths": "^3.8.0",
    "tslib": "^2.0.0",
    "webpack-cli": "3.3.6",
    "zone.js": "~0.11.4"
  },
```

**Comment by WeslleyDeSouza** [comment]
> Updating ng-universal v5.0.0 returns a strange error after trying to open angular page.
> 
> ng-universal v5.0.0 depend on nestjs 8 -> rxjs 7
> but angular 12 using rxjs 6 now.
> so we have a multi-level tree with cross dependencies
> 
> if using ng-universal v4 with angular 12 and @nguniversal/* 12, we have a multi-level tree with cross dependencies too.
> 
> as a result builder can not resolve dependency correctly.
> 
> Can we create an intermediate version for angular 12 with nestjs 7 and rxjs 6?

```
"dependencies": {
    "@angular/animations": "~12.1.4",
    "@angular/common": "~12.1.4",
    "@angular/compiler": "~12.1.4",
    "@angular/core": "~12.1.4",
    "@angular/forms": "~12.1.4",
    "@angular/platform-browser": "~12.1.4",
    "@angular/platform-browser-dynamic": "~12.1.4",
    "@angular/platform-server": "~12.1.4",
    "@angular/router": "~12.1.4",
    "@nestjs/common": "^8.0.5",
    "@nestjs/core": "^8.0.5",
    "@nestjs/ng-universal": "^5.0.0",
    "@nestjs/passport": "^8.0.0",
    "@nestjs/platform-express": "^8.0.5",
    "@nguniversal/common": "12.1.0",
    "@nguniversal/express-engine": "12.1.0",
    "class-transformer": "^0.2.3",
    "class-validator": "^0.9.1",
    "classlist.js": "1.1.20150312",
    "core-js": "^2.6.5",
    "passport": "^0.4.1",
    "passport-local": "^1.0.0",
    "rxjs": "~6.6.7",
    "terser": "^4.1.2",
    "tsconfig-paths": "^3.8.0",
    "tslib": "^2.0.0",
    "webpack-cli": "3.3.6",
    "zone.js": "~0.11.4"
  },
```

**Comment by ScuroGuardiano** [comment]
I saw on log that Angular 12 can use RxJS 7.x.x too
![obraz](https://user-images.githubusercontent.com/16851923/139760035-2f013713-8185-48ca-bf67-c9db5e4ad1e2.png)

So I changed in `package.json` RxJS from `~6.6.0` to `^7.0.0`
Then after installing I noticed another problem. NestJS in package json was in version `^7.0.0` which required RxJS in version 6. So I had to change
```
    "@nestjs/common": "^7.0.0" -> "^8.0.0",
    "@nestjs/core": "^7.0.0 -> "^8.0.0",
    "@nestjs/platform-express": "^7.0.0" -> "^8.0.0",
```

And it worked.
